docs/aws/audit/esmonitoring/rules/es_upgrade_available.mdx

```diff
@@ -24,6 +24,212 @@ AWSWAF, SOC2, NISTCSF
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. Make sure you have the necessary permissions to access the ElasticSearch domains.
+
+2. Once the AWS CLI is set up, you can list all the ElasticSearch domains using the following command:
+
+   ```
+   aws es list-domain-names
+   ```
+   This command will return a list of all the ElasticSearch domains.
+
+3. To get the detailed configuration of each domain, use the following command:
+
+   ```
+   aws es describe-elasticsearch-domain --domain-name your_domain_name
+   ```
+   Replace 'your_domain_name' with the actual domain name from the list obtained in the previous step. This command will return a detailed configuration of the specified ElasticSearch domain.
+
+4. In the output of the above command, look for the 'ElasticsearchVersion' field. This field indicates the version of the ElasticSearch service software used by the domain. If the version is not the latest, it indicates a misconfiguration. You need to manually compare this version with the latest version available on the official ElasticSearch website.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you can start writing the script, you need to install the necessary Python libraries. For this task, you will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary libraries and establish a session: After installing the necessary libraries, you need to import them into your script and establish a session with AWS.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Connect to the ElasticSearch service: Now that you have a session, you can connect to the ElasticSearch service. You can do this by creating a client for the service.
+
+```python
+es_client = session.client('es')
+```
+
+4. Check the ElasticSearch domains for the latest service software: Now that you have a client for the ElasticSearch service, you can use it to list all of your ElasticSearch domains and check if they are using the latest service software.
+
+```python
+response = es_client.list_domain_names()
+
+for domain in response['DomainNames']:
+    domain_info = es_client.describe_elasticsearch_domain(DomainName=domain['DomainName'])
+    if not domain_info['DomainStatus']['ServiceSoftwareOptions']['UpdateAvailable']:
+        print(f"The domain {domain['DomainName']} is using the latest service software.")
+    else:
+        print(f"The domain {domain['DomainName']} is not using the latest service software.")
+```
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>